packages/design-system/src/components/web-components/ds-tooltip/ds-tooltip.test.tsx

```diff
@@ -0,0 +1,152 @@
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import './ds-tooltip';
+import './ds-tooltip-icon';
+
+jest.mock('@popperjs/core');
+
+const defaultProps = {
+  children: <ds-tooltip-icon data-testid="ds-tooltip-icon" />,
+  'class-name': 'ds-c-tooltip__trigger-icon',
+  title: 'Tooltip body content',
+  id: '1234',
+};
+
+const customTooltipText = 'Custom tooltip title text for our web component!';
+const customHeadingText = 'Custom tooltip heading text for our web component!';
+const childText = 'Bog standard child content.';
+
+const propsWithSlots = {
+  children: (
+    <>
+      <div slot="title">{customTooltipText}</div>
+      <div slot="content-heading">{customHeadingText}</div>
+      {childText}
+    </>
+  ),
+};
+
+function renderTooltip(customProps = {}) {
+  const props = { ...defaultProps, ...customProps };
+  return render(<ds-tooltip {...props} />);
+}
+
+describe('ds-tooltip', function () {
+  it('renders a tooltip', () => {
+    const { asFragment } = renderTooltip();
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('renders title and contentHeading when passed in as slots', () => {
+    renderTooltip(propsWithSlots);
+    const tooltipTrigger = screen.getByText(customTooltipText);
+    expect(tooltipTrigger).toHaveTextContent(customTooltipText);
+    expect(tooltipTrigger).toHaveTextContent(customHeadingText);
+  });
+
+  it('renders default trigger icon', () => {
+    renderTooltip();
+    const triggerEl = screen.queryByTestId('ds-tooltip-icon');
+    expect(triggerEl).toMatchSnapshot();
+  });
+
+  it('renders custom trigger component', () => {
+    renderTooltip({
+      component: 'a',
+      children: childText,
+    });
+    const triggerEl = screen.queryByText(childText);
+    expect(triggerEl).toMatchSnapshot();
+  });
+
+  it('renders inverse tooltip', () => {
+    const { asFragment } = renderTooltip({ inversed: 'true' });
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('renders dialog tooltip', () => {
+    renderTooltip({ dialog: 'true', children: childText });
+    const tooltipTrigger = screen.getByText(childText);
+    userEvent.click(tooltipTrigger);
+    const contentEl = screen.queryByRole('dialog');
+    expect(contentEl).not.toBeNull();
+    expect(contentEl).toMatchSnapshot();
+  });
+
+  it('closes tooltip when trigger focus is lost', async () => {
+    jest.useFakeTimers();
+    const { container } = renderTooltip();
+    const tooltip = container.querySelector('.ds-c-tooltip');
```

Comment:
I doubt this class name is going to change anytime soon which is why I didn't go through the trouble of assigning a testID for this query.